### ISSUE TITLE
Mesure de la réponse de référence en fréquence et en phase

### ISSUE THREAD

**Issue by jpcima** [jujudusud/BPD#4]
Utiliser la nouvelle version se trouvant ici : https://github.com/jpcima/amplifier-profiling
construction avec `qmake && make`
Testé avec une multitude de greffons et ça sort des résults propres et exploitables.

**Comment by jujudusud** [comment]
Salut @jpcima ,
Je vais essayer ça immédiatement :-) je reviens vers toi avec les résultats.

**Comment by jujudusud** [comment]
essayé et adopté !!! superbe bouzin ! :-)

**Comment by jpcima** [comment]
Les fichiers de mesure actuels :
[treble-all-range.tar.gz](https://github.com/jujudusud/BPD/files/2636785/treble-all-range.tar.gz)
[basse-all-range.tar.gz](https://github.com/jujudusud/BPD/files/2636786/basse-all-range.tar.gz)
On ne tient compte que de "lo". Les mesures "hi" ne sont pas utiles et souffrent de clipping.
À faire : les médiums.
(peut-on réouvrir l'issue tant que la problématique n'est pas close ?)

Voici quelques tracés ; j'ai pas mis plus que 5 courbes sur chacun.

Afin de recalculer le filtre il faudra au préalable éliminer les bandes parasites.
Sur la réponse fréquentielle on peut observer deux bosses l'une variable et l'autre qui est statique, et on peut voir que la partie statique n'est pas tout à fait calée sur le 0 dB.

Soit il faut identifier le paramétrage matériel qui met la réponse en situation la plus neutre, soit faire une étape de nettoyage après l'acquisition de données.

## Les basses

`set logscale x; plot "b0/lo.dat" u 1:(amp2db($2)) w lines t "B0", "b2/lo.dat" u 1:(amp2db($2)) w lines t "B2", "b5/lo.dat" u 1:(amp2db($2)) w lines t "B5", "b8/lo.dat" u 1:(amp2db($2)) w lines t "B8", "b10/lo.dat" u 1:(amp2db($2)) w lines t "B10"`

![capture du 2018-12-02 10-12-59](https://user-images.githubusercontent.com/17614485/49337889-e5377680-f61a-11e8-945f-ba473b950ab0.png)

## Les aigues

`set logscale x; plot "t0/lo.dat" u 1:(amp2db($2)) w lines t "T0", "t2/lo.dat" u 1:(amp2db($2)) w lines t "T2", "t4/lo.dat" u 1:(amp2db($2)) w lines t "T4", "t8/lo.dat" u 1:(amp2db($2)) w lines t "T8", "t10/lo.dat" u 1:(amp2db($2)) w lines t "T10"`

![capture du 2018-12-02 10-14-37](https://user-images.githubusercontent.com/17614485/49337903-1c0d8c80-f61b-11e8-9a26-5ba893e58555.png)

**Comment by jujudusud** [comment]
Me voilà de retour avec le fichiers de mesure pour les mediums.
Il y a un potard gradué de 0 à 10 et un sélecteur de fréquence d'action 220-450-800-1.6k-3kHz
[mid-all-range.tar.gz](https://github.com/jujudusud/BPD/files/2637290/mid-all-range.tar.gz)

sur cet ampli on voit aussi des switches bright, ultra high, ultra low dont j'ai mesuré l'écart avec les potentiometres à 5 et la frequence mid à 800Hz.
[switches.tar.gz](https://github.com/jujudusud/BPD/files/2637291/switches.tar.gz)

**Comment by jpcima** [comment]
## Les médiums (Fc=800 Hz)

`gnuplot> set logscale x; plot "m0-f800/lo.dat" u 1:(amp2db($2)) w lines t "M0", "m2-f800/lo.dat" u 1:(amp2db($2)) w lines t "M2", "m4-f800/lo.dat" u 1:(amp2db($2)) w lines t "M4", "m8-f800/lo.dat" u 1:(amp2db($2)) w lines t "M8", "m10-f800/lo.dat" u 1:(amp2db($2)) w lines t "M10"`

![capture du 2018-12-03 00-12-30](https://user-images.githubusercontent.com/17614485/49346277-346bbe80-f690-11e8-8586-07bc2de6bbf6.png)

**Comment by jpcima** [comment]
## Basses+Médiums+Hautes (Fc=800, K=4)

`set logscale x; plot "m4-f800/lo.dat" u 1:(amp2db($2)) w lines t "M4", "b4/lo.dat" u 1:(amp2db($2)) w lines t "B4", "t4/lo.dat" u 1:(amp2db($2)) w lines t "T4"`

![capture du 2018-12-03 00-45-53](https://user-images.githubusercontent.com/17614485/49346578-d4c3e200-f694-11e8-8488-492f8bd09ba4.png)

**Comment by jpcima** [comment]
## Effet switch appliqué à B5

`set logscale x; plot "switch-ultra-low/lo.dat" u 1:(amp2db($2)) w lines t "Ultra-Low", "switch-ultra-high/lo.dat" u 1:(amp2db($2)) w lines t "Ultra-High", "switch-bright/lo.dat" u 1:(amp2db($2)) w lines t "Bright", "b5/lo.dat" u 1:(amp2db($2)) w lines t "B5"`

![capture du 2018-12-03 00-57-56](https://user-images.githubusercontent.com/17614485/49346705-80b9fd00-f696-11e8-9513-f2a9619d55bb.png)

**Comment by jujudusud** [comment]
Je vais commencer par mon analyse sur les basses.

Il y a un filtre passe haut à 40Hz avec une pente environ -15 dB / Octave.
Cela correspond à ce qu'il faut filtrer pour une basse 4 cordes en accordage standard.
Le filtre doit être déplacé à 30Hz pour filtrer un basse 5 cordes en accordage standard.

Ensuite, il y a en série à ce filtre un passe bande à 40Hz aussi, avec un Q=1 ou autour de 1, +10dB / -20dB.
Je pense qu'il faudra tester si ces valeurs sont à adapter à la 5 cordes en poussant un peu vers le bas, 30Hz, la fréquence d'action.

**Comment by jujudusud** [comment]
J'ai ajouté dans /filters les fichiers pour un filtre butterworth actif :
- Filtre actif passe-haut 40Hz - 12 dB/octave - Butterworth:  
- Les condensateurs sont choisis arbitrairement. Je prends C0= C1-C2= 150nF. Les deux constantes sont les mêmes, et C0 étant égal à 159155/R0*Fc, alors RO= 159155/C0*Fc.

- R0= 159155/150nF*40Hz= 28.52K.
- R1= 28.52/1.414= 37.51K 
- R2= 28.52/0.707= 75.02K 

la compilation faust plante. mais j'ai peut etre fait une erreur quelque part dans les valeurs...

**Comment by jpcima** [comment]
> J'ai ajouté dans /filters les fichiers pour un filtre butterworth actif.

Ce n'est pas nécessaire de passer par la discrétisation de circuit analogique pour construire le filtre de Butterworth. Celui ci se construit assez facilement en numérique et il existe une multitude d'implémentations.

**Comment by jujudusud** [comment]
Non, mais j'ai fais un patch pour tester et il plante... je voulais saoir si tu arrivais à quelque chose avec lui ou si c'est un bug ...
Pour le filtre butterworth, j'ai trouvé aussi dans 'iemlib' tout ce qu'il faut! :-)
